test/runtests.jl

```diff
@@ -47,8 +45,23 @@ if GROUP == "All" || GROUP == "AD"
     to_posdef_diagonal(a::AbstractVector) = Diagonal(a.^2 .+ 1)
 
     # Create vectors in probability simplex.
-    to_simplex(x::AbstractArray) = NNlib.softmax(x; dims=1)
+    function to_simplex(x::AbstractArray)
+        max = maximum(x; dims=1)
+        y = exp.(x .- max)
+        y ./= sum(y; dims=1)
+        return y
+    end
     to_simplex(x::AbstractArray{<:AbstractArray}) = to_simplex.(x)
+    function to_simplex_pullback(ȳ::AbstractArray, y::AbstractArray)
+        x̄ = ȳ .* y
+        x̄ .= x̄ .- y .* sum(x̄; dims=1)
+        return x̄
+    end
+    function ChainRulesCore.rrule(::typeof(to_simplex), x::AbstractArray{<:Real})
+        y = to_simplex(x)
+        pullback(ȳ) = (NoTangent(), to_simplex_pullback(ȳ, y))
```

Comment:
May want to think about a `ProbectTo` here, but that is beyond scope of this PR I guess